src/Tribe/Linked_Posts.php

```diff
@@ -1120,21 +1124,48 @@ public function saved_linked_post_dropdown( $post_type, $current = null ) {
 		// Get the label to use in placeholder attrs.
 		$label = $this->get_create_or_find_labels( $post_type, $creation_enabled );
 
+		$render_option = function ( $option ) {
+			?>
+			<option <?php selected( $option['selected'] ) ?> value="<?php echo esc_attr( $option['id'] ); ?>">
+				<?php echo esc_html( $option['text'] ); ?>
+			</option>
+			<?php
+		};
+
 		if ( $linked_posts || $my_linked_posts ) {
-			echo '<input
-				type="hidden"
-				class="tribe-dropdown linked-post-dropdown"
-				name="' . esc_attr( $name ) . '"
-				id="saved_' . esc_attr( $post_type ) . '"
-				data-placeholder="' . $label . '"
-				data-search-placeholder="' . $label . '" ' .
-				( $creation_enabled ?
-				'data-freeform
+
+			?>
+			<select
+				class="tribe-dropdown linked-post-dropdown hide-before-select2-init"
+				name="<?php echo esc_attr( $name ); ?>"
+				id="saved_<?php echo esc_attr( $post_type ); ?>"
+				data-placeholder="<?php echo esc_attr( $label ); ?>"
+				data-search-placeholder="<?php echo esc_attr( $label ); ?>"
+				<?php if ( $creation_enabled ) : ?>
+				data-freeform
 				data-sticky-search
-				data-create-choice-template="' . __( 'Create: <b><%= term %></b>', 'the-events-calendar' ) . '" data-allow-html ' : '' ) .
-				'data-options="' . esc_attr( json_encode( $data ) ) . '"' .
-				( empty( $current ) ? '' : ' value="' . esc_attr( $current ) . '"' ) .
-			'>';
+				data-create-choice-template="<?php echo __( 'Create: <b><%= term %></b>', 'the-events-calendar' ); ?>"
```

Comment:
* :no_entry_sign: All output should be run through an escaping function (see the Security sections in the WordPress Developer Handbooks), found '__'.
* :warning: Found Underscore.js unescaped output notation: "<%=".
* :warning: Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')'

<sub>hash: 120b1dffa199a1498d6b196f3d17fdb7</sub>